src/fetchBaseQuery.ts

```diff
@@ -1,28 +1,73 @@
 import { QueryApi } from './buildThunks';
+import { isPlainObject, joinUrls } from './utils';
 
 interface FetchArgs extends RequestInit {
   url: string;
+  params?: URLSearchParams | Record<string, string>;
+  body?: any;
+  responseHandler?: 'json' | 'text' | ((response: Response) => Promise<any>);
+  validateStatus?: (response: Response, body: any) => boolean;
 }
 
-export function fetchBaseQuery({ baseUrl }: { baseUrl: string } = { baseUrl: '' }) {
+const defaultValidateStatus = (response: Response) => response.status >= 200 && response.status <= 299;
+
+const isJsonContentType = (headers: Headers) => headers.get('content-type')?.trim()?.startsWith('application/json');
```

Comment:
Probably should also handle there I had forgotten about:

application/ld+json
application/vnd.api+json

Reply:
Just leaving a note for posterity: this would only matter if we used this to inspect the headers on the response, which we currently do not do. We did discuss having an 'auto' option for `responseHandler` that would make a best guess based on the incoming response header, but as of right now that's probably overkill.